addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/NeeoRoomHandler.java

```diff
@@ -0,0 +1,289 @@
+                        protocol.refreshScenarioName(channelKey);
+                        break;
+                    case NeeoBindingConstants.CHANNEL_CONFIGURED:
+                        protocol.refreshScenarioConfigured(channelKey);
+                        break;
+                    case NeeoBindingConstants.CHANNEL_STATUS:
+                        protocol.refreshScenarioStatus(channelKey);
+                        break;
+                }
+                break;
+        }
+    }
+
+    /**
+     * Simply cancels any existing initialization tasks and schedules a new task
+     *
+     * @see org.eclipse.smarthome.core.thing.binding.BaseThingHandler#initialize()
+     */
+    @Override
+    public void initialize() {
+        NeeoUtil.cancel(initializationTask.getAndSet(scheduler.schedule(new Runnable() {
+            @Override
+            public void run() {
+                initializeTask();
+            }
+        }, 1, TimeUnit.SECONDS)));
+    }
+
+    /**
+     * Initializes the task be creating the {@link NeeoRoomProtocol}, going online and then scheduling the refresh task.
+     */
+    private void initializeTask() {
+        final NeeoRoomConfig config = getConfigAs(NeeoRoomConfig.class);
+        try {
+            NeeoUtil.checkInterrupt();
+            final NeeoBrainApi brainApi = getNeeoBrainApi();
+            if (brainApi == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                return;
+            }
+
+            NeeoUtil.checkInterrupt();
+            final NeeoRoomProtocol protocol = new NeeoRoomProtocol(
+                    new StatefulHandlerCallback(new NeeoHandlerCallback() {
+
+                        @Override
+                        public void statusChanged(ThingStatus status, ThingStatusDetail detail, String msg) {
+                            updateStatus(status, detail, msg);
+                        }
+
+                        @Override
+                        public void stateChanged(String channelId, State state) {
+                            updateState(channelId, state);
+                        }
+
+                        @Override
+                        public void setProperty(@NonNull String propertyName, String propertyValue) {
+                            getThing().setProperty(propertyName, propertyValue);
+                        }
+
+                        @Override
+                        public void scheduleTask(Runnable task, long milliSeconds) {
+                            scheduler.schedule(task, milliSeconds, TimeUnit.MILLISECONDS);
+                        }
+                    }), brainApi, config.getKey());
+            NeeoUtil.close(roomProtocol.getAndSet(protocol));
+
+            NeeoUtil.checkInterrupt();
+            updateStatus(ThingStatus.ONLINE);
+
+            NeeoUtil.checkInterrupt();
+
+            NeeoUtil.cancel(refreshTask.getAndSet(scheduler.scheduleAtFixedRate(new Runnable() {
```

Comment:
May not apply here, but `scheduleWithFixedDelay` is prefered, see: https://github.com/openhab/static-code-analysis/issues/62